api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,418 @@
+)
+
+// TemplateRef is used to specify the installation CR templates
+type TemplateRef struct {
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// NodeSpec
+type NodeSpec struct {
+	// BmcAddress holds the URL for accessing the controller on the network.
+	// +required
+	BmcAddress string `json:"bmcAddress,omitempty"`
+
+	// BmcCredentialsName is the name of the secret containing the BMC credentials (requires keys "username" and "password").
+	// +required
+	BmcCredentialsName BmcCredentialsName `json:"bmcCredentialsName,omitempty"`
+
+	// Which MAC address will PXE boot? This is optional for some
+	// types, but required for libvirt VMs driven by vbmc.
+	// +kubebuilder:validation:Pattern=`[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}`
+	// +required
+	BootMACAddress string `json:"bootMACAddress,omitempty"`
+
+	// When set to disabled, automated cleaning will be avoided during provisioning and deprovisioning.
+	// Set the value to metadata to enable the removal of the disk’s partitioning table only, without fully wiping the disk. The default value is disabled.
+	// +optional
+	// +kubebuilder:default:=disabled
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode bmh_v1alpha1.AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
+
+	// RootDeviceHints specifies the device for deployment.
+	// Identifiers that are stable across reboots are recommended, for example, wwn: <disk_wwn> or deviceName: /dev/disk/by-path/<device_path>
+	// +optional
+	RootDeviceHints bmh_v1alpha1.RootDeviceHints `json:"rootDeviceHints,omitempty"`
+
+	// NodeNetwork is a set of configurations pertaining to the network settings for the node.
+	// +optional
+	NodeNetwork aiv1beta1.NMStateConfigSpec `json:"nodeNetwork,omitempty"`
+
+	// NodeLabels allows the specification of custom roles for your nodes in your managed clusters.
+	// These are additional roles are not used by any OpenShift Container Platform components, only by the user.
+	// When you add a custom role, it can be associated with a custom machine config pool that references a specific configuration for that role.
+	// Adding custom labels or roles during installation makes the deployment process more effective and prevents the need for additional reboots
+	// after the installation is complete.
+	// +optional
+	NodeLabels map[string]string `json:"nodeLabels,omitempty"`
+
+	// Hostname is the desired hostname for the host
+	// +required
+	HostName string `json:"hostName,omitempty"`
+
+	//TODO: remove NodeIp
+	NodeIp string `json:"nodeIp,omitempty"`
+
+	// Provide guidance about how to choose the device for the image being provisioned.
+	// +kubebuilder:default:=UEFI
+	BootMode bmh_v1alpha1.BootMode `json:"bootMode,omitempty"`
+
+	// UserData holds the reference to the Secret containing the user data to be passed to the host before it boots.
+	// +optional
+	UserData corev1.SecretReference `json:"userData,omitempty"`
```

Comment:
From https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#userdata:
> A reference to the Secret containing the cloudinit user data and its namespace, so it can be attached to the host before it boots for configuring different aspects of the OS (like networking, storage, ...).

I retained `UserData` from the original SiteConfig definition, should we remove this?

Reply:
I don't know of any uses of it, but it is fine to leave it in.

Reply:
From what I can tell user data is only usable when you're not using an iso since both would be provided through virtual media.

@eranco74 is that your understanding after the conversation you had with the metal team?

Reply:
@eranco74 @carbonin should we remove this field and reintroduce it if required at a later stage?

Reply:
+1 @carbonin - I will remove it as we can always add it back later :)

Reply:
Removed - marking this conversation as resolved.